### PR TITLE
Do general refactor of the code

### PR DESCRIPTION
**Pull request by tranzystorekk**
Highlights:

- Use `f32::{clamp, min, max}` instead of custom fns
- Simplify nesting in some methods
- Rename getters `get_sth` to just `sth` in accordance with Rust API guidelines
- Auto impl for the `Segment` trait
- Switch snake collection to `VecDeque`
- Use `Itertools::find_position` instead of `.enumerate().find()`

### CODE REVIEW

**Review by mkulagowski — CHANGES_REQUESTED** [review]
Looks nice, but must be rebased before i can add any comments - code has changed since it was created ;d

File: src/game/renderer.rs [hunk 1]
```diff
@@ -69,8 +70,10 @@ impl Renderer {
 
     /// Draws whole `LineSnake` structure
     ///
-    pub fn draw_snake(ctx: &mut Context, snake: &LineSnake) {
-        snake.body.iter().for_each(|x| x.draw(ctx));
+    pub fn draw_snake(ctx: &mut Context, snake: &Snake) {
```
Comment (mkulagowski) [inline src/game/renderer.rs:73]: I liked foreach :)

Comment (tranzystorekk) [reply src/game/renderer.rs:73]: That's why I didn't change each and every one 😜

I can appreciate keeping a personal style to the code 😄

**Review by mkulagowski — APPROVED** [review]
After rebase it looks good, I can use it 👍 
Learned some new tricks from it as well!

**Pull request merged**